akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/TestProbe.scala

```diff
@@ -218,6 +220,22 @@ abstract class TestProbe[M] {
    */
   @InternalApi protected def expectMessageClass_internal[C](max: FiniteDuration, c: Class[C]): C
 
+  /**
+   * Same as `receiveN(n, remaining)` but correctly taking into account
+   * the timeFactor.
+   */
+  def receiveMessages(n: Int): JList[M] = receiveN_internal(n, getRemainingOrDefault.asScala).asJava
```

Comment:
had to use a different name in javadsl because TestProbeImpl extends both and it would conflict with different return types